upgrade/v122to123/upgrade.go

```diff
@@ -99,7 +99,7 @@ func upgradeBackups(namespace string, lhClient *lhclientset.Clientset) (err erro
 		backup.Status.URL = backupStatus.BackupURL
 		backup.Status.Error = backupStatus.Error
 		backup.Status.SnapshotName = backupStatus.SnapshotName
-		backup.Status.State = engineapi.ConvertEngineBackupState(backupStatus.State)
+		backup.Status.State = longhorn.BackupState(engineapi.ConvertEngineBackupState(backupStatus.State))
```

Comment:
engineapi.ConvertEngineBackupState return v1beta2.BackupState, but backup.Status.State is v1beta1.BackupState, so I do a type casting here.

Reply:
engineapi.ConvertEngineBackupState return v1beta2.BackupState, but backup.Status.State is v1beta1.BackupState, so I do type casting here.

Reply:
BTW, should the APIs in upgrade/vxxxtoxxx/upgrade.go be upgraded to v1beta2?
I'm not sure, so leave this to @shuo-wu's refactoring task.

Reply:
> engineapi.ConvertEngineBackupState return v1beta2.BackupState, but backup.Status.State is v1beta1.BackupState, so I do type casting here.

:thinking: Isn't this down upgrade from v1beta2 -> v1beta1?

Reply:
@jenting Yes. So upgrade/vxxxtoxxx/upgrade.go should use v1beta2 APIs to avoid the problem?
`upgrade/vxxxtoxxx/upgrade.go` is for v1beta1 upgrade, using v1beta2 APIs  is a bit weird.

Reply:
YES. We should use v1beta2 clientset in all `upgrade/vxxxtoxxx/upgrade.go` and you already did that.

Here we are confused why there is a type conversion `longhorn.BackupState()`.

Reply:
@shuo-wu  we don't need  type conversion after replacing the v1beta1 apis with v1beta2.

Reply:
Right. And I saw this modification was removed.